docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -11,410 +11,616 @@ The `ERC20.cairo` contract implements an approximation of https://eips.ethereum.
-    }
-
-    func decimals() -> (decimals: felt) {
-    }
-
-    func totalSupply() -> (totalSupply: Uint256) {
-    }
-
-    func balanceOf(account: felt) -> (balance: Uint256) {
-    }
-
-    func allowance(owner: felt, spender: felt) -> (remaining: Uint256) {
-    }
-
-    func transfer(recipient: felt, amount: Uint256) -> (success: felt) {
-    }
-
-    func transferFrom(sender: felt, recipient: felt, amount: Uint256) -> (success: felt) {
-    }
-
-    func approve(spender: felt, amount: Uint256) -> (success: felt) {
-    }
+[,rust]
+----
+#[starknet::interface]
+trait IERC20<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn decimals(self: @TState) -> u8;
+    fn total_supply(self: @TState) -> u256;
+    fn balance_of(self: @TState, account: ContractAddress) -> u256;
+    fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(
+        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
 }
 ----
 
 === ERC20 compatibility
 
-Although StarkNet is not EVM compatible, this implementation aims to be as close as possible to the ERC20 standard, in the following ways:
+Although Starknet is not EVM compatible, this implementation aims to be as close as possible to the ERC20 standard, in the following ways:
 
-* It uses Cairo's `uint256` instead of `felt`.
-* It returns `TRUE` as success.
-* It accepts a `felt` argument for `decimals` in the constructor calldata with a max value of 2{caret}8 (imitating `uint8` type).
+* It uses Cairo's `u256` instead of `felt252`.
 * It makes use of Cairo's short strings to simulate `name` and `symbol`.
 
 But some differences can still be found, such as:
 
-* `transfer`, `transferFrom` and `approve` will never return anything different from `TRUE` because they will revert on any error.
-* Function selectors are calculated differently between https://github.com/starkware-libs/cairo-lang/blob/7712b21fc3b1cb02321a58d0c0579f5370147a8b/src/starkware/starknet/public/abi.py#L25[Cairo] and https://solidity-by-example.org/function-selector/[Solidity].
+* `transfer`, `transfer_from` and `approve` will never return anything different from `true` because they will revert on any error.
+* Function selectors are calculated differently between https://github.com/starkware-libs/cairo/blob/7dd34f6c57b7baf5cd5a30c15e00af39cb26f7e1/crates/cairo-lang-starknet/src/contract.rs#L39-L48[Cairo] and https://solidity-by-example.org/function-selector/[Solidity].
 
 == Usage
 
 Use cases go from medium of exchange currency to voting rights, staking, and more.
 
-Considering that the constructor method looks like this:
+Coming soon!
 
-[,cairo]
+== Extensibility
+
+(*This is currently unsafe*)
```

Comment:
```suggestion
:components: https://community.starknet.io/t/cairo-1-contract-syntax-is-evolving/94794#extensibility-and-components-11[Components]

WARNING: This is currently unsafe, because storage members could clash among used contracts if not reviewed carefully. Extensibility will be revisited after {components} are introduced.
```

Reply:
<img width="765" alt="Screenshot 2023-08-14 at 13 27 35" src="https://github.com/OpenZeppelin/cairo-contracts/assets/26070224/2f2a3b8d-34ae-4251-80b6-fa412b89cbc3">